### PR TITLE
Expand canaries to new regions

### PR DESCRIPTION
**Pull request by sobolk**
## Problem

We're running canaries only in `us-west-2` and therefore we might be late to detect service roll outs that may impact us.

**Issue number, if available:**

## Changes

This PR has the following changes:
1. It adds three more regions to both canary checks and resource cleanups
2. Adds/fixes existing actions/scripts where `us-west-2` was hardcoded
3. Adds retry mechanism at process level that retries commands 2 times by default. This is to keep probability of false positive low while we expand horizontally.

**Corresponding docs PR, if applicable:**

## Validation

canaries: https://github.com/aws-amplify/amplify-backend/actions/runs/8526147426
cleanup: https://github.com/aws-amplify/amplify-backend/actions/runs/8526285215

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: scripts/retry.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  'Usage:\n./scripts/retry.js <command>\n./scripts.retry.js --maxAttempts <number> <command>';
+
+class RetryOptions {
+  constructor(maxAttempts, command) {
+    this.maxAttempts = maxAttempts;
+    this.command = command;
+  }
+}
+
+const parseInput = () => {
```
Comment (edwardfoyle) [inline scripts/retry.js:26]: We can use the built-in [parseArgs](https://nodejs.org/api/util.html#utilparseargsconfig) utility to make args parsing a little cleaner / more robust

**Review by edwardfoyle — COMMENTED** [review]
LGTM, just one suggestion for arg parsing

File: scripts/retry.js (hunk 1 above)
Comment (sobolk) [reply scripts/retry.js:26]: I'll try. If this can surface the vararg-ish tail (the command part) nicely then I'll use it.

Comment (sobolk) [reply scripts/retry.js:27]: https://github.com/aws-amplify/amplify-backend/pull/1233/commits/02a8f092dd1500d0ad01e28d2e35fa6e8a75048e

File: scripts/retry.js [hunk 2]
```diff
@@ -24,22 +25,24 @@ class RetryOptions {
 }
 
 const parseInput = () => {
-  if (process.argv.length < 3) {
+  try {
+    const parsedResults = parseArgs({
+      allowPositionals: true,
+      options: {
+        maxAttempts: {
+          type: 'string',
```
Comment (sobolk) [inline scripts/retry.js:33]: image

Comment (sobolk) [reply scripts/retry.js:33]: no number, we have to parse :(

**Review by edwardfoyle — APPROVED** [review]

File: scripts/retry.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env node
+
```
Comment (0618) [inline scripts/retry.js:2]: Why don't we use typescript for this file?

**Review by 0618 — APPROVED** [review]

File: scripts/retry.js (hunk 3 above)
Comment (sobolk) [reply scripts/retry.js:2]: This is explained in the comment below.

The problem is that to use typescript we have to run `npm install` first. However, there are cases where we'd like to use retry without installing dependencies. Including wrapping `npm install` command itself as it tend to flake from time to time due to network glitches.

Comment (sobolk) [reply scripts/retry.js:2]: The alternative that I was considering was a bash script. But I figured that dependency-less pure JS is easier for us to write/read/understand.

**Pull request merged**